### DISCUSSION TITLE
Pre-release extensions

### DISCUSSION THREAD

**Discussion by isidorn** [microsoft/vscode-discussions#2] [Announcements]
Pre-release extensions allow extension authors to have a pre-release version of their extension for easier testing and feedback - similar to what the VS Code team has with VS Code Insiders. 

More details can be found here https://code.visualstudio.com/api/working-with-extensions/publishing-extension#prerelease-extensions

Let us know if you have any questions or if we can help. Thanks

**Comment by ericdrobinson** [comment]
Until the documentation is updated with more information, some may find the release notes helpful as it provides an overview of what the Pre-release workflow looks like for end users. See:

https://code.visualstudio.com/updates/v1_63#_pre-release-extensions

**Comment by vaibhavsingh97** [comment]
I have a question, is there any possibility that the vscode marketplace in `vscode` will show the `Install pre-release` button by default if no release exists?

What I did is that I created a `vsix` file using the `--pre-release` flag and uploaded it to Azure Devops Dashboard. So whenever a user searches for our extension and hits the install button, it gives an error message that no release exists. Users have to select 'Install pre-release' from the dropdown, which is a bad UX IMO, I wanted to fix this experience for our end users.

Cc @isidorn

**Reply by isidorn** [reply]
I think this is a fair request, and the best would be to create a new issue in the VS Code repo.

**Reply by vaibhavsingh97** [reply]
Thanks for replying, will do

**Comment by wolf99** [comment]
Is there some way to stop vscode asking if I would like to switch to pre-release versions of an extension every time I start a new vscode window?

**Reply by r3m0t** [reply]
I think that's coming from the extension itself so file an issue there.

**Reply by isidorn** [reply]
Correct. Just curious, what extension is doing this?
